src/interface.h

```diff
@@ -0,0 +1,20 @@
+#include <R.h>
+#include <Rinternals.h>
+#include <Rversion.h>
+
+SEXP r_particle_alloc(SEXP r_create, SEXP r_update, SEXP r_free,
```

Comment:
How important is it to have both particle and dust interfaces exposed? Would dust with N = 1 be ok?

Reply:
Not hugely important, but it might help with testing. With N = 1 it should be the same, except that there will be an extra dimension on the matrix. We can drop it whenever it's more hassle than it was worth